### PR TITLE
[ENG-7158][ENG-7060][docs] add `resourceClass` field to eas json schema

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

https://linear.app/expo/issue/ENG-7158/add-resourceclass-to-build-profile
https://linear.app/expo/issue/ENG-7060/addupdate-resource-class-names

# How

Add information about new `resourceClass` field in eas json schema

# Test Plan

Test locally

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by dsokal — APPROVED** [review]

**Comment by dsokal** [comment]
Maybe it's worth mentioning resource classes on the infrastructure page? https://docs.expo.dev/build-reference/infrastructure/

**Review by brentvatne — APPROVED** [review]

**Review by dsokal — APPROVED** [review]

File: docs/public/static/schemas/unversioned/eas-json-build-common-schema.js [hunk 1]
```diff
@@ -48,6 +48,13 @@ export default [
       '[Learn more about custom development clients](../../clients/introduction).',
     ],
   },
+  {
+    name: 'resourceClass',
+    enum: ['default', 'medium'],
```
Comment (wkozyra95) [inline docs/public/static/schemas/unversioned/eas-json-build-common-schema.js:53]: if we have a medium class, why there is a default too? aren't those the same?

Comment (szdziedzic) [reply docs/public/static/schemas/unversioned/eas-json-build-common-schema.js:53]: @wkozyra95 We discussed it here https://exponent-internal.slack.com/archives/C02123T524U/p1673430449218979?thread_ts=1673429656.875129&cid=C02123T524U

Comment (szdziedzic) [reply docs/public/static/schemas/unversioned/eas-json-build-common-schema.js:53]: Currently they both point to the medium instances, however default can be moved to large instances in the future, but medium will always explicitly point to medium instances

**Comment by szdziedzic** [comment]
@dsokal I will create docs describing resource classes tomorrow in a separate PR. I would like to merge it today so the schema is up to date.

**Review by amandeepmittal — APPROVED** [review]

**Pull request merged**